### PR TITLE
Add creation date and deadline

### LINKED ISSUE #30: As a user, I can add a deadline for a project
... so that I know which project will be due soon.

### LINKED ISSUE #31: As a user, I can edit a deadline for a project.

### LINKED ISSUE #32: As a user, I can view the project creation date.

### LINKED ISSUE #33: Add deadline and project creation date

### PR DESCRIPTION
**Pull request by claracheong4**
Changed command word for add project from add to project to match user guide.

### CODE REVIEW

File: src/main/java/seedu/momentum/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -54,8 +59,34 @@ public static Name parseName(String name) throws ParseException {
...
+        String trimmedDescription = description.trim();
+        return new Description(trimmedDescription);
+    }
+
+    /**
+     * Parses {@code Optional<String> date} and {@code Optional<String> time}into a {@code Deadline}.
+     * Leading and trailing whitespaces will be trimmed.
+     */
+    public static Deadline parseDeadline(Optional<String> date, Optional<String> time) throws ParseException {
+        if (date.isEmpty() || date.get().isBlank()) {
```
Comment (boundtotheearth) [inline src/main/java/seedu/momentum/logic/parser/ParserUtil.java:71]: Should this throw an exception instead?

**Review by boundtotheearth — COMMENTED** [review]
Is it necessary to split Date and Time?

I think it would be best if there is just 1 field for the user to specify both date and time, and if the user dosen't put a time, then the app just puts a empty value.

e.g.
project n/ProjectName d/Description and
project n/ProjectName d/Description dl/2020-01-01 and
project n/ProjectName d/Description dl/2020-01-01 10:10:10
are all valid commands handled by the app appropriately

In terms of the actual classes themselves, would it be more convenient to just use LocalDateTime (if we don't care about the time then just put some random value) instead of 3 separate Date, TIme, DateTime classes

File: src/main/java/seedu/momentum/logic/parser/ParserUtil.java (hunk 1 above)
Comment (claracheong4) [reply src/main/java/seedu/momentum/logic/parser/ParserUtil.java:71]: This is to show deletion of a deadline so the deadline "created" is an empty one.

**Comment by claracheong4** [comment]
Time is compulsory for LocalDateTime in java.time so unless a dummy time is added in, LocalDateTime cannot be used.
I think it would be cumbersome to try to differentiate between the dummy time and the actual time if any. I think this mainly has to do with the backend part.
Then since the time and date is separated, it would easier to parse them separately.

**Review by boundtotheearth — APPROVED** [review]
Ok. I'm just worried about having to maintain 3 separate data structures for this. I'm also worried if this is code duplication since all 3 classes will have basically the same code.

Also, merging will be a pain, but its probably better in the long run otherwise it gets confusing

Parsing them seperately is easier, but from a user perspective I think it is easier for date and time to be entered under the same dd/ or dl/ tag. Can probably do this another time

**Comment by claracheong4** [comment]
I think I will create a general date superclass then make datetime, date and time inherit from it.

**Review by kkangs0226 — APPROVED** [review]
Just wondering if it would be better to keep to one deadline command to indicate both date and time (so we don't need dt/ and dd/) so if you want to indicate time, you can do something like dl/VALID_DATE, VALID_TIME and if you don't, you can just put dl/VALID_DATE. Perhaps we could discuss with the group. Other than that, LGTM:)

**Comment by kkangs0226** [comment]
> Just wondering if it would be better to keep to one deadline command to indicate both date and time (so we don't need dt/ and dd/) so if you want to indicate time, you can do something like dl/VALID_DATE, VALID_TIME and if you don't, you can just put dl/VALID_DATE. Perhaps we could discuss with the group. Other than that, LGTM:)

I just realised that you already addressed this issue, and I'm ok with what you have suggested! Thanks:)

**Comment by claracheong4** [comment]
After checking the hierarchy of java.time, I realised the only common class ancestor of Date, Time and DateTime is java.lang.Object but they do implement similar interfaces. So I decided to make Date, Time and DateTime to implement a common Instance interface instead of a common class. And also if Date and Time inherit from the same class, it would imply that Date and Time is comparable.

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -46,7 +51,7 @@
     private final EditProjectDescriptor editProjectDescriptor;
 
     /**
-     * @param index of the project in the filtered project list to edit
+     * @param index                 of the project in the filtered project list to edit
```
Comment (pr4aveen) [inline src/main/java/seedu/momentum/logic/commands/EditCommand.java:54]: Should there be a big space here?

**Review by pr4aveen — APPROVED** [review]
LGTM. 

As the rest mentioned, I would personally prefer date and time to be keyed in using the same command. Will be good if this can be implemented in the future

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java (hunk 2 above)
Comment (claracheong4) [reply src/main/java/seedu/momentum/logic/commands/EditCommand.java:54]: It's just the work of reformat code to align with the next @param

**Pull request merged**